### PR TITLE
Show last run time for each command

### PR DESCRIPTION
**Pull request by dmfay**
with thanks to [atuin](https://github.com/ellie/atuin) for the idea of showing timing!

### CODE REVIEW

File: src/history/history.rs [hunk 1]
```diff
@@ -244,13 +245,27 @@ impl History {
 
         like_query.push_str("%");
 
-        let query = "SELECT id, cmd, cmd_tpl, session_id, when_run, exit_code, selected, dir, rank,
+        let query = "WITH matched_commands AS (
+                           SELECT id, cmd, cmd_tpl, session_id, when_run, exit_code, selected, dir, rank,
```
Comment (cantino) [inline src/history/history.rs:249]: Does using this `when_run` not work, such that you need to do the second query?  If this isn't the most recent, maybe  when the contextual_commands temp table is made, it should be `MAX(when_run)`?

**Comment by cantino** [comment]
Could we make it use less space by just showing `5d` or `30m` or `10s`. I don't think `30m 10s` adds much?

File: src/history/history.rs (hunk 1 above)
Comment (dmfay) [reply src/history/history.rs]: There would need to be at least two queries here to get the max grouped only by `cmd` but on looking temp table creation is a _much_ better place for this!

**Comment by dmfay** [comment]
I've liked having days-hours and hours-minutes resolution in my test-driving so far, so I may have sold it a bit short with the recent git commands. That said, other scales are definitely not as useful, so that specific band really has to be worth the four characters. Try it out for a day or three and see what you think, maybe?

**Comment by cantino** [comment]
> I've liked having days-hours and hours-minutes resolution in my test-driving so far, so I may have sold it a bit short with the recent git commands. That said, other scales are definitely not as useful, so that specific band really has to be worth the four characters. Try it out for a day or three and see what you think, maybe?

Ok!

**Comment by cantino** [comment]
I don't mind the two part dates, but I am finding it a bit hard to parse when I skim the list. I wonder if we should make it a lighter color (on dark backgrounds) and put it flush to the right?

**Comment by dmfay** [comment]
here's a version with timings on the right. The FixedLengthGraphemeString looks like it only checks on input and returns whatever the internal string has accumulated (even if shorter), though, and I'm not quite sure what to do about that. We have `width` but do we have the current cursor x somewhere?

**Comment by cantino** [comment]
I'm not 100% sure what you're asking. If you need to know if the line is currently selected in `truncate_for_display`, you could pass in `index == self.selection` to it. Is that what you're asking? All FixedLengthGraphemeString does is ensure that you cannot add more than N unicode characters to the string.

Do you like how it looks if you put 

```
write!(screen, "{}", cursor::Goto(width - 9, index as u16 + RESULTS_TOP_INDEX)).unwrap();
```

on line 248?

**Comment by dmfay** [comment]
I was trying to fill the remaining width between command text and timing info with spaces and not having much luck. `cursor::Goto` does the trick, thanks!

**Comment by cantino** [comment]
Thanks @dmfay, this is a nice addition :)
